### PR TITLE
MONGOID-5267 remove driver argument in queryable constructor

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

**Comment by p-mongo** [comment]
The Marshal.dump/load api needs to be retained, meaning the serialized form should still be the 6-element array with `nil` being written instead of `driver`, and when loading the second element should be retrieved and ignored.

**Comment by p-mongo** [comment]
Actually let's write `:mongo` as Mongoid was previously doing.

**Comment by Neilshweky** [comment]
done @p-mongo

File: lib/mongoid/criteria/marshalable.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@ module Marshalable
       #
       # @return [ Array<Object> ] The dumped data.
       def marshal_dump
-        data = [ klass, driver, inclusions, documents, strategy, negating ]
+        data = [ klass, :mongo, inclusions, documents, strategy, negating ]
```
Comment (p-mongo) [inline lib/mongoid/criteria/marshalable.rb:14]: Yes, but the `load` method shouldn't set `@driver` anymore. Could you also add a note that `:mongo` is written here for backwards compatibility with Mongoid 7 and earlier serializations please?

Comment (Neilshweky) [reply lib/mongoid/criteria/marshalable.rb:17]: done and done

**Review by p-mongo — APPROVED** [review]
:+1:  Thank you!

**Pull request merged**